### ISSUE TITLE
[Vita] Can't rename characters

### ISSUE THREAD

**Issue by evillurker** [diasurgical/devilutionX#1414]
Naming characters is impossible on new version. You get the default name but you cannot delete it, the input shows empty while the name of the character "Kharazim" stays, and if you add anymore letters it says "Kharazim" and then w/e you input.
Didn't happen on older versions.

**Comment by AJenbo** [comment]
Can you add a screenshot of this (or photo of the device) during hero naming.

**Comment by evillurker** [comment]
https://ibb.co/vJfXCZ1
https://ibb.co/d7YDj8B
Basically the input you get is null.. so you can't actually "delete" anything
And whatever you add, gets added to "Kharazim"

**Comment by AJenbo** [comment]
@isage is this something you know how to solve?

**Comment by isage** [comment]
I'll take a look, but that looks like a bug in vita sdl2 port (looks like i messed some stuff migrating it from one sdk to another).

**Comment by evillurker** [comment]
Btw about how it used to work:
You would already get the "Kharazim" on the input, and then you would just delete it and add your own.

**Comment by AJenbo** [comment]
Grate, we are preparing a 1.2.1 release to solve a couple of issues on Windows so that would be a good time to get this out as well.

**Comment by isage** [comment]
Yeah, ok. That's not an issue in SDL2, but rather in devilution itself.
Vita only have on-screen keyboard, so SDL_StartTextInput emits only SDL_TEXTINPUT event. Sadly, there's no way to pre-set text for on-screen keyboard in SDL2.
And SDL_TEXTINPUT handler *appends* to character name.
On previous builds pre-generated names on vita were disabled, so there was nothing to append to and naming worked properly.

So there's 3 ways to fix that:
1. SDL_TEXTINPUT handler shouldn't append to name, but rather replace it.
2. Same, but only on vita
3. Disable pre-generated names on vita

**Comment by isage** [comment]
Well, there's also 4. Add (vita-specific) function on sdl side to allow setting OSK text.

**Comment by AJenbo** [comment]
ok, switch is in a similar situation could you check if a similar solution would work? Else I think 1 might be the solution. @glebm do you have an opinion on this from the perspective of other handhelds?

**Comment by isage** [comment]
I don't have switch, nor really familiar with it.
But as far as i can see, switch port doesn't use sdl for osk at all.

**Comment by AJenbo** [comment]
Yeah, it uses a switch-specific api, since SDL does not support prefilled input.

**Comment by isage** [comment]
I guess i can do the same for vita, but that'll take some time.

**Comment by AJenbo** [comment]
Ok, no rush, the issue isn't really critical.

**Comment by glebm** [comment]
> Sadly, there's no way to pre-set text for on-screen keyboard in SDL2.

I think it'd be better to actually add such a function to SDL, so that users can do:

```c++
#if SDL_VERSION_AT_LEAST(2, 0, 16)
if (SDL_HasScreenKeyboardSupport()) {
  SDL_SetScreenKeyboardText(initial_text);
}
#endif
SDL_StartTextInput();
```

**Comment by isage** [comment]
Yeah. I'll propose that to upstream.
Still requires 1, though.

**Comment by evillurker** [comment]
If you need testing for switch let me know
I also have that
Available on discord EvilLurker#3059
Thank you all for your help
